core/error.go

```diff
@@ -177,3 +177,20 @@ func IsValueTransferErr(e error) bool {
 	_, ok := e.(*ValueTransferError)
 	return ok
 }
+
+type BadHashErr struct {
+	Message string
+}
+
+func (err *BadHashErr) Error() string {
+	return err.Message
+}
+
+func BadHashError(hash common.Hash) error {
+	return &BadHashErr{Message: fmt.Sprintf("Found known bad hash in chain %x", hash)}
+}
+
+func IsBadHashErr(err error) bool {
+	_, ok := err.(*BadHashErr)
+	return ok
+}
```

Comment:
I'm not entirely sure (maybe @fjl can pinch in on this one), but it is very unusual to create a wrapper to check if an error is of a specific type or not. The Go libs AFAIK use the embedded check directly at the appropriate places, so I'd stick to that convention. But I may be wrong in this respect.